data/moves.js

```diff
@@ -5531,18 +5526,17 @@ exports.BattleMovedex = {
 		flags: {distance: 1},
 		onHitField: function (target, source) {
 			let targets = [];
-			for (let sideSlot = 0; sideSlot < this.sides.length; sideSlot++) {
-				let sideActive = this.sides[sideSlot].active;
-				for (let activeSlot = 0; activeSlot < sideActive.length; activeSlot++) {
-					if (sideActive[activeSlot] && sideActive[activeSlot].isActive && sideActive[activeSlot].hasType('Grass')) {
+			for (const sideSlot of this.sides) {
+				for (const activeSlot of sideSlot.active) {
+					if (activeSlot && activeSlot.isActive && activeSlot.hasType('Grass')) {
 						// This move affects every Grass-type Pokemon in play.
-						targets.push(sideActive[activeSlot]);
+						targets.push(activeSlot);
 					}
 				}
 			}
 			let success = false;
-			for (const target of targets) {
-				success = this.boost({def: 1}, target, source, this.getMove('Flower Shield')) || success;
+			for (const pokemon of targets) {
```

Comment:
i dont remember doing this but im already in bed so ill fix it tomorrow, feel free to point out anything else like this